### PR TITLE
Fixes clustergen output

### LINKED ISSUE #3314: Clustergen producing too many negative output, poor diffs
**Bug description**

clustergen test used in the provider tests needs is producing too many errors due to incorrect TKR versions, and not accounting for additional validation on the AWS side. Also CC-side clustergen needs to account for the appropriate TKR values in the injected TKR_DATA as well.

**Expected behavior**

positive test output more often than not, usable to evaluate before/after effects of PR being tested

**Steps to reproduce the bug**
check provider tests output

### PR DESCRIPTION
**Pull request by vuil**
### What this PR does / why we need it

to provider better diffs by generating more positive case output and using correct TKR values.

Also: capture different test results files based on infra type being tested, and added more documentation about the CC-based clustergen workflow.

Note : topmost commit "test clustergen" included for demonstration purposes to show diffs created, will be removed before this change is committed.

### Which issue(s) this PR fixes

Fixes #3314

### Describe testing done for PR

```
> make clustergen

> cd pkg/v1/providers/tests/clustergen/testdata

> cat new/resul* | cut -d: -f2 | sort | uniq -c
   3 # results
  20 NEG
  40 POS
```

### Release note

```release-note
None
```

### CODE REVIEW

File: pkg/v1/providers/infrastructure-aws/v1.2.0/yttcc/overlay.yaml [hunk 1]
```diff
@@ -55,6 +55,7 @@ spec:
         annotations:
           #! VVV TODO(vui) os-name handling
           run.tanzu.vmware.com/resolve-os-image: #@ "ami-region={},os-name=ubuntu".format(data.values.AWS_REGION)
+          tkg.tanzu.vmware.com/test-annotation: test
```
Comment (tenczar) [inline pkg/v1/providers/infrastructure-aws/v1.2.0/yttcc/overlay.yaml:58]: what is this used for?

File: pkg/v1/providers/ytt/09_miscellaneous/08_store_data_values.yaml [hunk 2]
```diff
@@ -44,6 +44,7 @@ metadata:
   labels:
     tkg.tanzu.vmware.com/cluster-name: #@ data.values.CLUSTER_NAME
     clusterctl.cluster.x-k8s.io/move: ""
+    tkg.tanzu.vmware.com/testnoncc: test
```
Comment (tenczar) [inline pkg/v1/providers/ytt/09_miscellaneous/08_store_data_values.yaml:47]: what is this used for?

**Review by tenczar — APPROVED** [review]
Two small comments. Are these annotations necessary? What is their purpose.

File: pkg/v1/providers/ytt/09_miscellaneous/08_store_data_values.yaml (hunk 2 above)
Comment (vuil) [reply pkg/v1/providers/ytt/09_miscellaneous/08_store_data_values.yaml:47]: please see the PR description. This is only included for demo purpose to show the resultant diffs, and will be removed before merging.

File: pkg/v1/providers/infrastructure-aws/v1.2.0/yttcc/overlay.yaml (hunk 1 above)
Comment (vuil) [reply pkg/v1/providers/infrastructure-aws/v1.2.0/yttcc/overlay.yaml:58]: please see the PR description. This is only included for demo purpose to show the resultant diffs, and will be removed before merging.

**Review by anujc25 — APPROVED** [review]
LGTM

**Pull request merged**